streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -369,10 +369,15 @@ public static StreamThread create(final TopologyMetadata topologyMetadata,
                                       final Runnable shutdownErrorHook,
                                       final BiConsumer<Throwable, Boolean> streamsUncaughtExceptionHandler) {
 
+        final boolean stateUpdaterEnabled = InternalConfig.stateUpdaterEnabled(config.originals());
+
         final String threadId = clientId + THREAD_ID_SUBSTRING + threadIdx;
+        final String stateUpdaterId = threadId.replace(THREAD_ID_SUBSTRING, STATE_UPDATER_ID_SUBSTRING);
+        final String effectiveThreadId = stateUpdaterEnabled ? stateUpdaterId : threadId;
```

Comment:
Could you please rename this variable to `restorationThreadId`?

Reply:
done